lib/utils/html_widgets.dart

```diff
@@ -0,0 +1,152 @@
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    Widget imageWidget = Image.network(
+      foundResource ?? "",
+      width: size.width,
+      height: size.height,
+    );
+
+    if (foundEventTrigger != null) {
+      return GestureDetector(
+        onTap: _handleTapEvent(foundEventTrigger),
+        child: imageWidget,
+      );
+    }
+
+    return imageWidget;
+  }
+
+  /// Builds custom Button widget from HTML [element] and applies [resources]
+  static Widget _buildCustomButtonWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    return ElevatedButton(
+        style: ButtonStyle(fixedSize: MaterialStateProperty.all(size)),
+        onPressed: foundEventTrigger != null
+            ? _handleTapEvent(foundEventTrigger)
+            : null,
+        child: Text(foundResource ?? ""));
+  }
+
+  /// Builds custom Video widget from HTML [element] and applies [resources]
+  static Widget? _buildCustomVideoWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? source = element.children
+        .firstWhereOrNull((element) => element.localName == "source")
+        ?.attributes[CustomHtmlWidgets.SRC];
+    String? foundResource =
+        resources.firstWhereOrNull((resource) => resource.id == source)?.data;
+
+    if (source != null) {
+      return Container(
+        width: size.width,
+        height: size.height,
+        child: VideoWidget(src: foundResource ?? ""),
+      );
+    }
+
+    return null;
+  }
+
+  /// Finds event trigger assigned to this [element] from [eventTriggers]
+  static ExhibitionPageEventTrigger? _findEventTrigger(
+    dom.Element element,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    String? elementId = element.id;
+    return eventTriggers.firstWhereOrNull(
+        (eventTrigger) => eventTrigger.clickViewId == elementId);
+  }
+
+  /// Event handler for tap events on custom widgets per [eventTrigger]
+  ///
+  /// TODO: Implement this
+  static Function() _handleTapEvent(ExhibitionPageEventTrigger eventTrigger) {
+    return () {};
+  }
+
+  /// Extracts elements width and height
+  static Size _extractSize(dom.Element element) {
```

Comment:
This seems to handle size as pixels, is it possible that size is sometimes handled as e.g. percents?

Reply:
Lets discuss this in person.